core/trino-main/src/main/java/io/trino/sql/analyzer/Field.java

```diff
@@ -169,4 +163,26 @@ public String toString()
 
         return result.toString();
     }
+
+    public static class OriginColumnDetail
```

Comment:
The `details` are distinct per expression, but if you want to reason about them further, e.g. collect all references from a query, this class will need `equals()`.